src/sentry/integrations/perforce/repository.py

```diff
@@ -48,51 +88,133 @@ def build_repository_config(
         Returns:
             Repository configuration
         """
+        depot_path = data["identifier"]
+
         return {
-            "name": "",
-            "external_id": "",
-            "url": "",
-            "config": {},
-            "integration_id": 0,
+            "name": depot_path,
+            "external_id": data["external_id"],
+            "url": f"p4://{depot_path}",
+            "config": {
+                "depot_path": depot_path,
+                "name": depot_path,
+            },
+            "integration_id": data["integration_id"],
         }
 
+    def _format_commits(
+        self, changelists: list[dict[str, Any]], depot_path: str, client: Any
+    ) -> Sequence[Mapping[str, Any]]:
+        """
+        Format Perforce changelists into Sentry commit format.
+
+        Args:
+            changelists: List of changelist dictionaries from P4
+            depot_path: Depot path
+            client: Perforce client instance
+
+        Returns:
+            List of commits in Sentry format
+        """
+        commits = []
+
+        # Cache user info to avoid multiple lookups for the same user
+        user_cache: dict[str, dict[str, Any] | None] = {}
+
+        for cl in changelists:
+            try:
+                # Handle potentially null/invalid time field
+                time_value = cl.get("time") or 0
+                try:
+                    time_int = int(time_value)
+                except (TypeError, ValueError):
+                    time_int = 0
```

Comment:
We need to report errors and understand why the data may be incorrect, otherwise, we will have bugs hard to debug down the line.

```suggestion
                    time_int = 0
                    logger.warning("some message")
```